### ISSUE TITLE
Toolkit: Plugin style checks don't run on Windows

### ISSUE THREAD

**Issue by gordonjb** [grafana/grafana#35674]
**What happened**:
When running a plugin build using `yarn build` that previously completed successfully on Windows on Linux, the build fails with linting issues.

Windows: 
![image](https://user-images.githubusercontent.com/2196521/121924497-dd9cd080-cd33-11eb-88fd-83733000c786.png)

Linux:
![image](https://user-images.githubusercontent.com/2196521/121924595-f5745480-cd33-11eb-8d34-597a24ab616e.png)

**What you expected to happen**:
Linting issues should be reported when running `yarn build` on Windows.

**How to reproduce it (as minimally and precisely as possible)**:
* Run `yarn build` on Windows
* Run `yarn build` on Linux

**Anything else we need to know?**:

**Environment**:
![image](https://user-images.githubusercontent.com/2196521/121924265-a5958d80-cd33-11eb-8567-c5b60890d680.png)

**Comment by mckn** [comment]
@gordonjb Could you please provide us with the actual linting error output? Thanks 🙏🏻

**Comment by gordonjb** [comment]
Apologies for the � 's, blame Jenkins encoding I guess, but local runs on the build machine exhibit the same output
```
$ grafana-toolkit plugin:build
  Using Node.js [object Object]
  Using @grafana/toolkit [object Object]
- Preparing
  Created: /plugindir/.prettierrc.js
��� Preparing
- Linting
  
  
  
  /plugindir/src/DataSource.ts
     73:17  error  Replace `results` with `(results)`                                                                                                                                                                                                                                                                                     prettier/prettier
     87:17  error  Replace `results` with `(results)`                                                                                                                                                                                                                                                                                     prettier/prettier
    101:17  error  Replace `results` with `(results)`                                                                                                                                                                                                                                                                                     prettier/prettier
    115:17  error  Replace `results` with `(results)`                                                                                                                                                                                                                                                                                     prettier/prettier
    274:9   error  Replace `target` with `(target)`                                                                                                                                                                                                                                                                                       prettier/prettier
    276:12  error  Replace `target` with `(target)`                                                                                                                                                                                                                                                                                       prettier/prettier
    330:15  error  Replace `target` with `(target)`                                                                                                                                                                                                                                                                                       prettier/prettier
    331:11  error  Replace `(target` with `((target)`                                                                                                                                                                                                                                                                                     prettier/prettier
    361:43  error  Replace `level` with `(level)`                                                                                                                                                                                                                                                                                         prettier/prettier
    409:14  error  Replace `(target` with `((target)`                                                                                                                                                                                                                                                                                     prettier/prettier
    410:12  error  Replace `target` with `(target)`                                                                                                                                                                                                                                                                                       prettier/prettier
    706:37  error  Insert `;`                                                                                                                                                                                                                                                                                                             prettier/prettier
    728:15  error  Expected an assignment or function call and instead saw an expression                                                                                                                                                                                                                                                  @typescript-eslint/no-unused-expressions
    728:26  error  Insert `;`                                                                                                                                                                                                                                                                                                             prettier/prettier
    742:64  error  Insert `;`                                                                                                                                                                                                                                                                                                             prettier/prettier
    761:12  error  Replace `��name:��'level',��type:��FieldType.other,��config:��{��displayNameFromDS:��'Level',��custom:��{displayMode:��'image',��width:��'1'}��}` with `���������������������������name:��'level',���������������������������type:��FieldType.other,���������������������������config:��{��displayNameFromDS:��'Level',��custom:��{��displayMode:��'image',��width:��'1'��}��},���������������������`  prettier/prettier
    788:73  error  Expected '!==' and instead saw '!='                                                                                                                                                                                                                                                                                    eqeqeq
    808:1   error  Delete `��`                                                                                                                                                                                                                                                                                                             prettier/prettier
    841:31  error  Expected '===' and instead saw '=='                                                                                                                                                                                                                                                                                    eqeqeq
    843:22  error  Expected '===' and instead saw '=='                                                                                                                                                                                                                                                                                    eqeqeq
    845:22  error  Expected '===' and instead saw '=='                                                                                                                                                                                                                                                                                    eqeqeq
  
  /plugindir/src/components/ExploreCheatSheet.tsx
     87:24  error  Replace `e` with `(e)`  prettier/prettier
    125:24  error  Replace `e` with `(e)`  prettier/prettier
  
  /plugindir/src/components/ExploreErrorsQueryEditor.tsx
      6:1   error  '@grafana/data' import is duplicated                                                                                                                                                                                                                                          no-duplicate-imports
     44:1   error  Delete `����`                                                                                                                                                                                                                                                                   prettier/prettier
     52:4   error  Delete `;`                                                                                                                                                                                                                                                                    prettier/prettier
     59:1   error  Delete `����`                                                                                                                                                                                                                                                                   prettier/prettier
     73:27  error  Replace `���������������������������width={9.5}���������������������������className="query-keyword"���������������������������tooltip={�����������������������������"Select��which��errors��will��be��returned."���������������������������}�����������������������` with `��width={9.5}��className="query-keyword"��tooltip={'Select��which��errors��will��be��returned.'}`  prettier/prettier
     89:23  error  Replace `value` with `(value)`                                                                                                                                                                                                                                                prettier/prettier
    106:25  error  Replace `e` with `(e)`                                                                                                                                                                                                                                                        prettier/prettier
    151:19  error  Replace `e` with `(e)`                                                                                                                                                                                                                                                        prettier/prettier
    163:19  error  Replace `e` with `(e)`                                                                                                                                                                                                                                                        prettier/prettier
    175:19  error  Replace `e` with `(e)`                                                                                                                                                                                                                                                        prettier/prettier
  
  /plugindir/src/components/ExploreLogsQueryEditor.tsx
      6:1   error  '@grafana/data' import is duplicated                                                  no-duplicate-imports
     75:75  error  Insert `��`                                                                            prettier/prettier
     78:32  error  Replace `query.levels��?��query.levels��:��[]` with `(query.levels��?��query.levels��:��[])`  prettier/prettier
    141:25  error  Replace `e` with `(e)`                                                                prettier/prettier
    142:31  error  Replace `e` with `(e)`                                                                prettier/prettier
    161:25  error  Replace `e` with `(e)`                                                                prettier/prettier
    216:19  error  Replace `e` with `(e)`                                                                prettier/prettier
    228:19  error  Replace `e` with `(e)`                                                                prettier/prettier
    240:19  error  Replace `e` with `(e)`                                                                prettier/prettier
    252:19  error  Replace `e` with `(e)`                                                                prettier/prettier
  
  /plugindir/src/components/ExploreNotesQueryEditor.tsx
     6:1   error  '@grafana/data' import is duplicated                                                                                                                                                                                no-duplicate-imports
    32:11  error  Delete `����`                                                                                                                                                                                                         prettier/prettier
    35:22  error  Replace `�������������������������������'Build��a��query��to��send��to��syst��from��a��list��of��available��systs.��Wildcards��accepted'���������������������������` with `'Build��a��query��to��send��to��syst��from��a��list��of��available��systs.��Wildcards��accepted'`  prettier/prettier
    60:25  error  Replace `e` with `(e)`                                                                                                                                                                                              prettier/prettier
    78:19  error  Replace `e` with `(e)`                                                                                                                                                                                              prettier/prettier
  
  /plugindir/src/components/ExploreQueryEditor.tsx
      6:1   error  '@grafana/data' import is duplicated  no-duplicate-imports
    131:19  error  Replace `e` with `(e)`                prettier/prettier
    143:19  error  Replace `e` with `(e)`                prettier/prettier
    155:19  error  Replace `e` with `(e)`                prettier/prettier
    167:19  error  Replace `e` with `(e)`                prettier/prettier
    179:19  error  Replace `e` with `(e)`                prettier/prettier
  
  /plugindir/src/components/NotesQueryEditor.tsx
    12:20  error  Delete `��`  prettier/prettier
  
  /plugindir/src/components/QueryEditor.tsx
    43:23  error  Replace `value` with `(value)`  prettier/prettier
  
  /plugindir/src/components/VariableQueryEditor.tsx
    41:21  error  Replace `v` with `(v)`  prettier/prettier
  
  ��� 57 problems (57 errors, 0 warnings)
    48 errors and 0 warnings potentially fixable with the `--fix` option.
  
  
  
  Trace: Error: 57 linting errors found in 16 files
      at /plugindir/node_modules/@grafana/toolkit/src/cli/tasks/plugin.build.js:118:31
      at step (/plugindir/node_modules/@grafana/toolkit/node_modules/tslib/tslib.js:140:27)
      at Object.next (/plugindir/node_modules/@grafana/toolkit/node_modules/tslib/tslib.js:121:57)
      at fulfilled (/plugindir/node_modules/@grafana/toolkit/node_modules/tslib/tslib.js:111:62)
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
      at /plugindir/node_modules/@grafana/toolkit/src/cli/utils/useSpinner.js:25:29
      at step (/plugindir/node_modules/@grafana/toolkit/node_modules/tslib/tslib.js:140:27)
      at Object.throw (/plugindir/node_modules/@grafana/toolkit/node_modules/tslib/tslib.js:121:57)
      at rejected (/plugindir/node_modules/@grafana/toolkit/node_modules/tslib/tslib.js:112:69)
      at processTicksAndRejections (internal/process/task_queues.js:97:5)
��� 57 linting errors found in 16 files
```

**Comment by mckn** [comment]
Do you have the same folder structure on both linux and windows? Looks like you have code below `/plugindir/src/components` that seems to not follow the linting rules.

**Comment by gordonjb** [comment]
The folder structure below the project directory is identical yes, it's the same Git repo checked out on both systems.

I'm aware that there is code that isn't following the linting rules, my issue is that those failures don't show up on Windows, where I do the plugin development. Linting issues are only revealed on the Linux build server.

**Comment by mckn** [comment]
Ok ok, super good feedback! I will look into this and try to see what is happening under the hood.

**Comment by gordonjb** [comment]
Thanks, sorry if I got a bit terse before coffee 😳 Let me know if you need any more info!

**Comment by bradenhs** [comment]
There may be multiple issues here but I looked into it and it appears that at least one of those issues has to do with the library [globby](https://github.com/sindresorhus/globby). In one instance where this library is used it fails to get a list of typescript files as expected.

https://github.com/grafana/grafana/blob/0db6ba9d707a9059deab633cd6351e245b99c114/packages/grafana-toolkit/src/cli/tasks/plugin.build.ts#L78

I verified that on Linux that function returns a list of files but on Windows it silently fails and returns an empty list. The upshot of this is that linting still runs on Windows but it lints a list of zero files and thus produces no errors :smile: Haven't looked at what the upstream issue may be exactly but I did see several issues in that globby repo about windows support. Perhaps using another library with better windows support would be a good option?
